### PR TITLE
Story/1934

### PR DESCRIPTION
**Pull request by ws-mps**

### CODE REVIEW

File: addons/udes_stock/models/stock_move.py [hunk 1]
```diff
@@ -156,11 +156,15 @@ def split_out_move_lines(self, move_lines):
                 'move_orig_ids': [],
                 'ordered_qty': total_ordered_qty,
                 'product_uom_qty': total_initial_qty,
-                'state': 'assigned',
+                'state': self.state,
+                # TODO: ^^^ Not always true if we split a move which
+                # is partially availabe
```
Comment (crwr45) [inline addons/udes_stock/models/stock_move.py]: This needs to be resolved.
Take a look at either `stock.move._get_relevant_state_among_moves()` or `stock.move._recompute_state()` from Odoo core stock module to see which is the most appropriate.

File: addons/udes_stock/models/stock_move.py [hunk 2]
```diff
@@ -156,11 +156,15 @@ def split_out_move_lines(self, move_lines):
...
             })
             move_lines.write({
                 'move_id': bk_move.id,
-                'state': 'assigned',
+                'state': self.state,
+                # TODO: ^^^ Not always true if we split a move which
+                # is partially availabe
```
Comment (crwr45) [inline addons/udes_stock/models/stock_move.py]: Same as previous comment.

File: addons/udes_stock/models/stock_move.py [hunk 3]
```diff
@@ -443,9 +454,13 @@ def refactor_action_group_by_move_line_key(self):
...
                     group_moves |= move
 
-            Picking._new_picking_for_group(key, group_moves)
+            origin = None
+            if len(group_pickings) == 1:
+                origin = group_pickings.origin
+            Picking._new_picking_for_group(key, group_moves, origin=origin)
             result_moves |= group_moves
 
+        # NOTE: empty pick = no moves and no move_lines
```
Comment (crwr45) [inline addons/udes_stock/models/stock_move.py]: I don't think this comment is necessary, the filtered should be self explanatory.

File: addons/udes_stock/models/stock_move.py [hunk 4]
```diff
@@ -330,7 +334,7 @@ def _create_moves_for_push(self, push, move_lines):
                 # TODO: reservation to the src location of a move if it has
                 # TODO: preceeding moves.
                 # TODO: Uncomment this, and unskip tests, when story/1934 is completed
-                # 'move_orig_ids': [(6, 0, [move.id,])]
+                'move_orig_ids': [(6, 0, [move.id,])]
```
Comment (crwr45) [inline addons/udes_stock/models/stock_move.py]: Doesn't this also then need `move_dest_ids` to be set on the previous move, as with the related change from https://taiga.udes.io/project/udescore/task/2030?kanban-status=87?

File: addons/udes_stock/models/stock_move.py [hunk 5]
```diff
@@ -384,7 +388,12 @@ def refactor_action_group_by_move_key(self):
                       '') % (key, move_group.mapped('location_id'),
                              move_group.mapped('location_dest_id')))
 
-            Picking._new_picking_for_group(key, move_group)
+            group_pickings = move_group.mapped('picking_id')
+            origin = None
+            if len(group_pickings) == 1:
```
Comment (crwr45) [inline addons/udes_stock/models/stock_move.py]: This should set the origin if there are multiple group picks with the same origin. Something like:
```
origins = set(group_pickings.mapped('origin'))
origin = None
if len(origins) == 1:
    origin = origins[0]

```

File: addons/udes_stock/models/stock_move.py [hunk 6]
```diff
@@ -443,9 +454,13 @@ def refactor_action_group_by_move_line_key(self):
                 else:
                     group_moves |= move
 
-            Picking._new_picking_for_group(key, group_moves)
+            origin = None
+            if len(group_pickings) == 1:
+                origin = group_pickings.origin
+            Picking._new_picking_for_group(key, group_moves, origin=origin)
```
Comment (crwr45) [inline addons/udes_stock/models/stock_move.py]: Again, keep the origin if all are the same.

File: addons/udes_stock/models/stock_move.py (hunk 3 above)
Comment (ws-mps) [reply addons/udes_stock/models/stock_move.py]: The filtered is wrong, we should fix it.

Comment (ws-mps) [reply addons/udes_stock/models/stock_move.py]: Just ignore my comment misread move_lines by move_line_ids

File: addons/udes_stock/models/stock_move.py (hunk 4 above)
Comment (nocturnalastro) [reply addons/udes_stock/models/stock_move.py]: I think we only need to set the one as they are related fields, when I checked it was linked already

File: addons/udes_stock/models/stock_move.py [hunk 7]
```diff
@@ -326,11 +334,7 @@ def _create_moves_for_push(self, push, move_lines):
             move_vals = base_vals.copy()
             move_vals.update({
                 'product_uom_qty': sum(mls.mapped('qty_done')),
-                # FIXME: CWR: This is commented out because odoo does not limit
-                # TODO: reservation to the src location of a move if it has
-                # TODO: preceeding moves.
-                # TODO: Uncomment this, and unskip tests, when story/1934 is completed
-                # 'move_orig_ids': [(6, 0, [move.id,])]
+                'move_orig_ids': [(6, 0, move.ids)]
```
Comment (crwr45) [inline addons/udes_stock/models/stock_move.py]: Very pleased to see that TODO resolved.

**Pull request merged**